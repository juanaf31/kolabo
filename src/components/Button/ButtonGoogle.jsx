import React from 'react'

function ButtonGoogle(props) {
    return (
        <div data-testid='google_btn' className="google-btn" {...props}>
            <div className="google-icon-wrapper">
                <img data-testid="google_image" className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" />
            </div>
            <p data-testid='text' className="btn-text"><b>{props.title}</b></p>
        </div>
    )
}

export default ButtonGoogle
