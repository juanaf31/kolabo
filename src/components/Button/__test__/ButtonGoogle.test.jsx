import { cleanup, getByText, render } from '@testing-library/react'
import React from 'react'
import ButtonGoogle from '../ButtonGoogle'
import renderer from 'react-test-renderer'

afterEach(() => {
    cleanup()
})

describe('testing button google', () => {
    const val = 'anyValue'

    test('render component correnctly', () => {

        const { getByTestId } = render(<ButtonGoogle />)

        getByTestId('google_btn')
        getByTestId('google_image')
        getByTestId('text')
    })

    test('render title correctly', () => {
        const { getByText } = render(<ButtonGoogle title={val} />)

        getByText(val)
    })

    test('render all possible props', () => {
        const { getByTestId } = render(<ButtonGoogle title={val} className={val} />)

        const google_btn = getByTestId('google_btn')

        expect(google_btn).toHaveClass(val)
        getByTestId('text')

    })

    test('button matches snapshot', () => {
        const tree = renderer.create(<ButtonGoogle />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})