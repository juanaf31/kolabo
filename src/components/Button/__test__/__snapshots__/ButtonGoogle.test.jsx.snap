// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing button google button matches snapshot 1`] = `
<div
  className="google-btn"
  data-testid="google_btn"
>
  <div
    className="google-icon-wrapper"
  >
    <img
      className="google-icon"
      data-testid="google_image"
      src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
    />
  </div>
  <p
    className="btn-text"
    data-testid="text"
  >
    <b />
  </p>
</div>
`;
